src/test/java/dev/karolkoltun/expenses/HomeFinancesServiceTest.java

```diff
@@ -0,0 +1,329 @@
+import dev.karolkoltun.exceptions.InvalidPayment;
+import dev.karolkoltun.exceptions.NoExpenseInCategoryException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.*;
+
+import static dev.karolkoltun.expenses.Category.*;
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+class HomeFinancesServiceTest {
+
+    private HomeFinancesService homeFinancesService;
+    private PlainCurrencyService plainCurrencyService;
+    private PlainAccountService plainAccountService;
+
+    @BeforeEach
+    void setup() {
+        plainCurrencyService = new PlainCurrencyService();
+        plainAccountService = new PlainAccountService();
+        homeFinancesService = new HomeFinancesService(plainCurrencyService, plainAccountService);
+    }
+
+    @Test
+    void shouldAddExpenseInPLN() throws InvalidPayment {
+        //Given
+        Expense expense = new Expense(LocalDate.now().minusDays(1), new BigDecimal(345), "CSK", OTHERS);
+
+        //When
+        homeFinancesService.addExpense(expense);
+        List<Expense> actualExpense = homeFinancesService.getAllExpenses();
+        String where = actualExpense.get(0).getWhere();
+        Category actualCategory = actualExpense.get(0).getCategory();
+
+        //Then
+        assertThat(actualExpense)
+                .as("List should contain added expense")
+                .contains(expense);
+
+        Expense addedExpense = actualExpense.get(0);
+
+        assertThat(addedExpense)
+                .extracting("where", "amount")
+                .containsExactly("CSK", BigDecimal.valueOf(345));
+
+    }
+
+    @Test
+    void shouldNotAddExpenseWithFutureDate() throws InvalidPayment {
+        //Given
+        Expense expense = new Expense(LocalDate.now().plusDays(3), new BigDecimal(120), "Rossmann", GIFTS);
+
+        //When
+        homeFinancesService.addExpense(expense);
+        List<Expense> actualExpense = homeFinancesService.getAllExpenses();
+
+        //Then
+        assertFalse(actualExpense.contains(expense));
+    }
+
+    @Test
+    void shouldNotAddExpenseWithNegativeAmount() throws InvalidPayment {
+        //Given
+        Expense expense = new Expense(LocalDate.now().minusDays(3), new BigDecimal(-45), "Rossmann", GIFTS);
+
+        //When
+        homeFinancesService.addExpense(expense);
+        List<Expense> actualExpense = homeFinancesService.getAllExpenses();
+
+        //Then
+        assertTrue(actualExpense.isEmpty());
+    }
+
+    @Test
+    void shouldAddExpenseInForeignCurrency() throws InvalidPayment {
+        // Given
+        CurrencyService currencyServiceMock =
```

Comment:
Tak jak napisałem wyżej, we wszystkich testach stosowałbym mocków obu pomocniczych serwisów `CurrencyService`. Za to konfigurację mocka `CurrencyService` zostawiłbym w tym teście bo jest rzeczą specyficzną dla tego testu.